bayem/vba.py

```diff
@@ -332,7 +332,9 @@ def run(self):
             i_iter += 1
 
             self.update_parameters(k, J)
-
+            
+            k, J = self.p(self.m)
```

Comment:
So apparently moving that evaluation of `k` and `J` reduces the number of iterations. Is that enough of a justification? Over in [`vary_avb`](https://github.com/physimals/vaby_avb/blob/master/vaby_avb/avb.py#L324), they do this evaluation within `cost_free_energy`, so right before calculating `F`... 
Maybe that is why I also (without much thought) moved it to after the noise update.

Reply:
To my eye , we deal with a set of nonlinear algebraic equations, which we are trying to solve by the fixed-point method. Generally, when we have a set of equations that are coupled/dependent to each other, it makes sense to account for this dependence/coupling in the middle of one whole iteration of fixed-point method; i.e. (in our case here) after updating parameters we update "k" and "J".
An important justification for this IMO is that, "k" and "J" are NOT dependent on noise parameters, so, updating "k" and "J" before the update of noises will not influence new values of "k" and "J", but can improve the update of noises (since we use more updated "J" and "k" for that).